pkg/operator/ceph/cluster/osd/osd_test.go

```diff
@@ -473,6 +473,48 @@ func TestOSDPlacement(t *testing.T) {
 	assert.Equal(t, "label2", result.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms[0].MatchExpressions[0].Key)
 }
 
+func TestDetectCrushLocation(t *testing.T) {
+	location := []string{"host=foo"}
+	nodeLabels := map[string]string{}
+
+	// no change to the location if there are no labels
+	updateLocationWithNodeLabels(&location, nodeLabels)
+	assert.Equal(t, 1, len(location))
+	assert.Equal(t, "host=foo", location[0])
+
+	// no change to the location if an invalid label or invalid topology
+	nodeLabels = map[string]string{
+		"topology.rook.io/foo":          "bar",
+		"invalid.topology.rook.io/rack": "r1",
+		"topology.rook.io/zone":         "z1",
+	}
+	updateLocationWithNodeLabels(&location, nodeLabels)
+	assert.Equal(t, 1, len(location))
+	assert.Equal(t, "host=foo", location[0])
+
+	// update the location with valid topology labels
+	nodeLabels = map[string]string{
+		"failure-domain.beta.kubernetes.io/region": "region1",
+		"failure-domain.beta.kubernetes.io/zone":   "zone1",
```

Comment:
Should we update these with the latest non-beta versions while we are here? Maybe it would be good to leave one beta and one non-beta?

Reply:
Both beta and stable topology labels are used in the unit test, so it seems ok as is until we stop supporting the beta labels. This particular example just uses the beta labels.

Reply:
I must have missed the other test. 👍